From 43dd9f43b1631e6188be9e3dc4d97f29273f8291 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Fri, 27 Jul 2018 14:46:53 +0300
Subject: [PATCH 397/448] staging: fsl-dpaa2/ethsw: Add ndo_bridge_set/getlink ops

These ndo ops are needed on kernel 4.9 in order to offload
things like learning and flooding, otherwise settings are
applied on the software bridge only.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
---
 drivers/staging/fsl-dpaa2/ethsw/ethsw.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/ethsw/ethsw.c b/drivers/staging/fsl-dpaa2/ethsw/ethsw.c
index 9182046..5694099 100644
--- a/drivers/staging/fsl-dpaa2/ethsw/ethsw.c
+++ b/drivers/staging/fsl-dpaa2/ethsw/ethsw.c
@@ -537,6 +537,8 @@ static netdev_tx_t port_dropframe(struct sk_buff *skb,
 	.ndo_change_mtu		= port_change_mtu,
 	.ndo_has_offload_stats	= port_has_offload_stats,
 	.ndo_get_offload_stats	= port_get_offload_stats,
+	.ndo_bridge_setlink     = switchdev_port_bridge_setlink,
+	.ndo_bridge_getlink     = switchdev_port_bridge_getlink,
 
 	.ndo_start_xmit		= port_dropframe,
 };
-- 
1.7.1

